indico/modules/events/management/controllers.py

```diff
@@ -328,6 +400,89 @@ def _process(self):
         return jsonify_data(flash=False)
 
 
+class RHClonePreview(RHManageEventBase):
+    def _process(self):
+        form = CloneRepeatabilityForm()
+
+        if form.repeatability.data == 'interval':
+            clone_calculator = IntervalCloneCalculator(self.event_new)
+        elif form.repeatability.data == 'pattern':
+            clone_calculator = PatternCloneCalculator(self.event_new)
+        else:
+            raise BadRequest
+
+        try:
+            dates = clone_calculator.calculate(request.form)
```

Comment:
Yes it can. Why is it an overkill? There are 2 value errors that might happen, one is wrong data in the form and the other is out of range data (>100). How'd you do it instead?